### ISSUE TITLE
[macOS] Should config and data values be swapped?

### ISSUE THREAD

**Issue by gtrrz-victor** [sindresorhus/env-paths#21]
Based on: https://developer.apple.com/library/archive/documentation/General/Conceptual/MOSXAppProgrammingGuide/AppRuntime/AppRuntime.html

<img width="1638" alt="Screenshot 2024-07-02 at 4 55 50 PM" src="https://github.com/sindresorhus/env-paths/assets/881031/6ee162ce-b9f6-437e-ae77-94a20df032ef">

I reckon that the library is returning swapped mac values. 
- `config` should return `data` value 
`~/Library/Application Support/MyApp-nodejs` instead of  `~/Library/Preferences/MyApp-nodejs`
- `data` should return `config` value
 `~/Library/Preferences/MyApp-nodejs` instead of `~/Library/Application Support/MyApp-nodejs` 

Thanks

**Comment by gtrrz-victor** [comment]
In that way, it would be aligned with the Go implementation: https://pkg.go.dev/os#UserConfigDir

**Comment by gtrrz-victor** [comment]
PR open with the fix: https://github.com/sindresorhus/env-paths/pull/22

**Comment by sindresorhus** [comment]
This is not correct. `config` could technically also have been the `~/Library/Application Support` directory, but `~/Library/Preferences` is more specific. It would be completely wrong to make `data` be `~/Library/Preferences` though.
